### PR TITLE
Resource level permissions

### PR DESCRIPTION
**Pull request by jnwaletzko**

### CODE REVIEW

File: lib/rolypoly/role.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+module Rolypoly
+  class Role
+    attr_accessor :name, :resource_type, :resource_id
+
+    def initialize(role)
```
Comment (jnwaletzko) [inline lib/rolypoly/role.rb]: I could use some idea's here. Basically what I'm trying to do is be able to pass. `(:financial_billing)` and/or `(financial_billing: :organization)`. If it's the second then set the `resource_type`.

File: lib/rolypoly/role.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+module Rolypoly
+  class Role
+    attr_accessor :name, :resource_type, :resource_id
```
Comment (jnwaletzko) [inline lib/rolypoly/role.rb]: oops, that's leftover from the initial way I was doing it. Good catch.

Comment (vivekbisen) [inline lib/rolypoly/role.rb]: So, does the check only know about the type of role? As in, what's the benefit of verifying that the role I wanna compare against is restricted to organization. Is there some other check that will determine whether this role with particular organization_id has restricted data access?

Comment (jnwaletzko) [inline lib/rolypoly/role.rb]: That's what the role_resource is. Basically on the controller there will be a method like
``` 
private def role_resource
  { organization: params[:organization_id] }
end 
```
And that's how the id is checked. The `resource_type` here is to determine if we need need to do that check. `role_object.resource?(role_resource)` checks to see if the resource_type and resource_id match the role.

**Pull request closed without merging**